### ISSUE TITLE
Unable to pull custom.list from Primary pihole, and gravity-sync.log does not exist.

### ISSUE THREAD

**Issue by rustafur** [vmstan/gravity-sync#95]
**Describe the Issue**

Upon my initial run of gravity-sync I ran across two issues.

1. Gravity-sync is unable to pull custom.list from primary pihole
2. gravity-sync.log does not exist

Running gravity-sync.sh compare for the first time.  I am running this from my secondary pihole server (hostnname piHoleTwo), and it does appear to successfully connect to my primary pihole server (IP 10.1.0.10)
![gravity-sync-compare](https://user-images.githubusercontent.com/3969850/93922989-5a84a780-fcd8-11ea-9259-311504cdf209.png)

Executing gravity-sync.sh for the first time
![gravity-sync-firstRun](https://user-images.githubusercontent.com/3969850/93923027-66706980-fcd8-11ea-82f7-a12fa7a318c1.png)

I went to check the log to see what I could find, and found that it's not there.
![gravity-sync-checkLogs](https://user-images.githubusercontent.com/3969850/93923080-8011b100-fcd8-11ea-9a97-ad3f56cab674.png)

![logcheck](https://user-images.githubusercontent.com/3969850/93923092-84d66500-fcd8-11ea-956d-0150fa6e0a92.png)

**Configuration:**
 - OS for Primary/Secondary: [Ubuntu 20.04.1 LXC for both]
 - Platform for Primary/Secondary: [Proxmox 6.2-11]
 - SSH Server/Client: [OpenSSH]
 - SSH Authentication: [Keypair]
 - Gravity Sync Version: [2.2.1]
 - Pi-hole Versions: [5.1.2]

**Additional Context**
I'm not doing anything extra or fancy, and I followed the instructions explicitly from the main github page.  Very nicely written instructions, by the way :)

**Comment by rv10guy** [comment]
I am seeing exactly the same thing.

**Comment by valantur** [comment]
@rv10guy @rustafur 

1) Edit the gravity-sync.sh script and go to line 223. Delete ".backup" from the rsync line so that it looks like this:

<img width="565" alt="Screen Shot 2020-10-01 at 12 33 29 PM" src="https://user-images.githubusercontent.com/2516392/94854825-5c3e2180-03e2-11eb-9434-b0b749fdfc15.png">

2) run the script again
3) profit

**Comment by rv10guy** [comment]
Perfect...Fixed...Thanks!

**Comment by vmstan** [comment]
OK so that is interesting, because it should be creating that .backup file by creating a copy of the existing file, just prior to executing the RSYNC.

**Comment by rv10guy** [comment]
If I look into my backup directory I have the following..

`2020-10-01-145730-gravity.db.backup
custom.list.pull
ravity.db.pull`

**Comment by vmstan** [comment]
@rv10guy can you do a `./gravity-sync.sh update` and run a sync it again? This should override your change, but it repositions where the backup of the custom.list file happens after the checks to make sure that it exists at the other side.

**Comment by rv10guy** [comment]
Okay so after an update all appears good. Which seems strange because that reversed the change I made to line 223. I tried several changes to the whitelist and custom DNS with success every time. Not sure what to say.

**Comment by rv10guy** [comment]
I do now have a 2020-10-01-151716-custom.list.backup which is new

**Comment by vmstan** [comment]
I just flipped around where it does the checking for the existence of the remote custom.list. I'm not sure why it worked for me (and others) and not for some people, but it still works for me and works for you now 👍

**Comment by rv10guy** [comment]
Thanks for your help!

**Comment by vmstan** [comment]
Actually I do now why now that I look at this again, so when it was running the old way, it would only backup the remote file IF there was an existing local file. It would then check to see if the base of the remote file existed, and then if so, attempt to backup the copy of that file. But the backup didn't exist because the backup itself never ran, because there was no local copy to tell it that it was needed to be overridden.

Basically it worked if you had used the script before, but new installs  that just doesn't work, unless you already have a copy of made by hand or make a dummy copy.

**Comment by valantur** [comment]
good stuff! thanks @vmstan

**Comment by old-square-eyes** [comment]
I get this error too. But when I run When I run...

```
$ gravity-sync/gravity-sync.sh update
[ INFO ] Gravity Sync 2.2.1 Executing
[  OK  ] Importing gravity-sync.conf Settings
[  OK  ] Evaluating Arguments: UPDATE Requested
fatal: not a git repository (or any of the parent directories): .git
[ FAIL ] Updating Cache
```
